### DISCUSSION TITLE
Gauge: Fixed number of digits (e.g. 1,00 €)

### DISCUSSION THREAD

**Discussion by StephanSchuster** [home-assistant/frontend#13510] [Configuration]
Problem: Gauge strips trailing 0s and displays e.g. 1,3 € instead of 1,30 €.
Request: Provide an option to ensure a fixed amount of digits, 2 in above case.

Original post: https://community.home-assistant.io/t/fixed-number-of-digits-in-official-gauge-card-e-g-1-00/404713

**Comment by bcutter** [comment]
So glad I'm not the only one being disrupted of this visualization. Possibility of setting the number count would be great so everyone can decide individually how many filler zeros are needed to have a consistent look and feel.

**Comment by Loic691** [comment]
As in the entity cards, we can adjust the number of digit by the
format: precisionX code with X equals to 0, 1,2,...
eg precision1
It would we great

**Comment by NdS-Research-Facilities** [comment]
Yes please, this is very annoying visually, as the font also changes size when the trailing zero is removed.

**Comment by vingerha** [comment]
Would be great to have

**Comment by mhspelt** [comment]
Upvoted, but perhaps I can make the feature request just a little more general: wouldn't it be great if the display value could be a template?

So that you are able to get the correct precision with

`display_value: {{ value | round(2) }}`

But you can also use it more generally. For example, I have a sensor whose value is a number of hours (as float) that I would like to have in a gauge, but as the text in the gauge I would like to see not "8.5235126512 h" but "8:32:56", for example. With the display value template, I could configure a gauge on the numeric sensor and then add

`display_value: {{ (value * 3600) | int | as_timestamp }}` 

for the displayed text.

**Reply by vingerha** [reply]
I now too voted for this but I am not sure if the whole voting in this thread becomes misty... who wants/what ???
Also not even sure if this thread is being read/measured wrt to votes correctly

**Reply by bcutter** [reply]
Basically good idea. I just don't think this will be a "GUI option" for normal users. Templating must still be seen as some kind of "advanced stuff" and is therefore consequently ignored in the UI.

So if it's your idea (which I would love to see) or a simple "show number of digits behind `.`/`,`" option: I'd go for the last one. Just because I feel like possibility to get this is higher than for the more flexible option.

**Reply by Loic691** [reply]
I think we have the 2 purposes :
- precisonX to be able to change it in frontend.
- display_value is great too to be able to change text.

+1 for the 2 items 😀

**Reply by mhspelt** [reply]
Good points made in the comments. Do you think it is better to split this to a separate request?

**Comment by vingerha** [comment]
BTW... yesterday I managed to get fixed digits with the custom card-templater (hacs)
based on a input_helper for testing so EXAMPLE
```
type: custom:card-templater
card:
  type: gauge
  entity: input_number.sunsetmin
entities:
  - entity: input_number.sunsetmin
    state_template: '{{ states("input_number.sunsetmin") | int(0) }}'
```

**Reply by mhspelt** [reply]
Thanks! I even managed to more-or-less solve my "format a timespan" issue this way:
```
state_template: >-
      {{ "{:0}.{:02}".format( 
               states("sensor.day_length") | int, 
               ( ( states("sensor.day_length") | float * 60 ) % 60 ) | int(0) ) }}
```
I would prefer to see "8:04" for 8 hours and 4 minutes, but it now shows "8.04 h" and that is good enough. At least better than "8,066666666666667 h" crammed into a tiny font label.

**Comment by bqzero** [comment]
That is exactly what I am looking for a long time. Also wanted to place a feature requedt for it.
It is very anoying qhen you get large numbers from an energy meter in W and there is no simple way to have it displayed with lets say 3 digitis and if necessary adjusted unit of meassure. No use for me to get 5.78GWh presented as 5780235.13W in the frontend. Not a nice display and too much unnecessary information. I mean it is nice to have this precision to work in the backend, but absolutely no need tonpresent it like that to users.
